keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -72,98 +73,84 @@ class GPT2CausalLM(Task):
-        "gpt2_base_en",
-        sequence_length=128,
-    )
+    features = ["The quick brown fox jumped.", "I forgot my homework."]
+    gpt2_lm = keras_nlp.models.GPT2CausalLM.from_preset("gpt2_base_en")
+    gpt2_lm.fit(x=features, batch_size=2)
+    ```
+
+    Call `fit()` without preprocessing.
+    ```python
+    x = {
+        "token_ids": tf.constant([[50256, 1, 2, 3, 4]] * 2),
+        "padding_mask": tf.constant([[1, 1, 1, 1, 1]] * 2),
+    }
+    y = tf.constant([[1, 2, 3, 4, 50256]] * 2)
+    sw = tf.constant([[1, 1, 1, 1, 1]] * 2)
 
-    # Create a `GPT2CausalLM`, using pretrained GPT2 and custom preprocessor.
     gpt2_lm = keras_nlp.models.GPT2CausalLM.from_preset(
         "gpt2_base_en",
-        preprocessor=preprocessor,
+        preprocessor=None,
     )
-    gpt2_lm.predict(["You know this is still a test string"])
+    gpt2_lm.fit(x=x, y=y, sample_weight=sw, batch_size=2)
     ```
 
-    Fit your preprocessed data with randomly initialized GPT2. This is useful
-    when you want to do data preprocessing inside `tf.data` pipeline.
+    Custom backbone and vocabulary.
     ```python
-    # Define preprocessed input.
-    features = {
-        "token_ids": tf.constant(
-            [[1, 2, 3, 4, 0, 0]] * 2, shape=(2, 6)
-        ),
-        "padding_mask": tf.constant(
-            [[1, 1, 1, 1, 0, 0]] * 2, shape=(2, 6)
-        ),
-    }
-    labels = tf.constant(
-        [[2, 3, 4, 0, 0, 0]] * 2, shape=(2, 6)
+    features = ["a quick fox.", "a fox quick."]
+    vocab = {"<|endoftext|>": 0, "a": 4, "Ġquick": 5, "Ġfox": 6}
+    merges = ["Ġ q", "u i", "c k", "ui ck", "Ġq uick"]
+    merges += ["Ġ f", "o x", "Ġf ox"]
+
+    tokenizer = keras_nlp.models.GPT2Tokenizer(
+        vocabulary=vocab,
+        merges=merges,
     )
-    sample_weight = tf.constant(
-        [[1, 1, 1, 0, 0, 0]] * 2, shape=(2, 6)
+    preprocessor = keras_nlp.models.GPT2CausalLMPreprocessor(
+        tokenizer=tokenizer,
+        sequence_length=128,
     )
-
-    # Randomly initialize a GPT2 backbone.
     backbone = keras_nlp.models.GPT2Backbone(
-        vocabulary_size=50257,
-        num_layers=2,
-        num_heads=2,
-        hidden_dim=128,
-        intermediate_dim=256,
+        vocabulary_size=30552,
+        num_layers=4,
+        num_heads=4,
+        hidden_dim=256,
+        intermediate_dim=512,
         max_sequence_length=128,
     )
-    # Create a `GPT2CausalLM` without preprocessor and fit the data.
-    gpt2_lm = keras_nlp.models.GPT2CausalLM(backbone, preprocessor=None)
-    gpt2_lm.compile(
-        loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
-    )
-    gpt2_lm.fit(
-        x=features,
-        y=labels,
-        sample_weight=sample_weight,
-        batch_size=2,
+    gpt2_lm = keras_nlp.models.GPT2CausalLM(
+        backbone=backbone,
+        preprocessor=preprocessor,
     )
+    gpt2_lm.generate("I want to say", max_length=30)
```

Comment:
Good call. I am going to switch to showing `fit` for this bring your own data example. Otherwise we would be showing generate on a random backbone, which would have garbage results anyway.